pkg/client/userd/trafficmgr/session.go

```diff
@@ -566,34 +567,37 @@ func (s *session) getInfosForWorkloads(
 		}
 		for _, workload := range wls {
 			if _, ok := wiMap[workload.GetUID()]; ok {
+				if _, ok := wiMap[workload.GetUID()].Services[string(svc.UID)]; !ok {
+					wiMap[workload.GetUID()].Services[string(svc.UID)] = &rpc.WorkloadInfo_ServiceReference{
```

Comment:
Why do three hash lookups of the same value?
```suggestion
			if wlInfo, ok := wiMap[workload.GetUID()]; ok {
				if _, ok := wlInfo.Services[string(svc.UID)]; !ok {
					wlInfo.Services[string(svc.UID)] = &rpc.WorkloadInfo_ServiceReference{
```